packages/twenty-front/src/modules/object-metadata/components/ObjectMetadataItemsLoadEffect.tsx

```diff
@@ -21,6 +34,10 @@ export const ObjectMetadataItemsLoadEffect = () => {
       skip: !isLoggedIn,
     });
 
+  const filteredNewObjectMetadataItems = useMemo(() => {
+    return filterTsVectorFields(newObjectMetadataItems);
+  }, [newObjectMetadataItems]);
+
   const [objectMetadataItems, setObjectMetadataItems] = useRecoilState(
```

Comment:
we don't need to listen on objectMetadataItems here as we only use it to conditionnally setObjectMetadataItems.

What we could do is to use a useRecoilCallback which enables us to do the setObjectMetadataItems and to check at the last moment what we have in objectMetadataItems without having the component listening to it (and therefore re-rendering)